nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/stream/StreamServiceImpl.java

```diff
@@ -225,4 +239,200 @@ public void beforeStop(ComponentManager mgr, boolean isStandby) {
             Framework.getRuntime().getComponentManager().removeListener(this);
         }
     }
+
+    // ---------------------------------------------------------------
+    // Debug tools for internal use
+
+    /**
+     * Display up to {@code limit} record from a Log.
+     */
+    public void cat(String logConfig, String name, int limit) throws Exception {
+        cat(logConfig, name, limit, this::renderMessage);
+    }
+
+    public void cat(String logConfig, String name, int limit, Consumer<LogRecord<Record>> render) throws Exception {
+        try (LogTailer<Record> tailer = getLogManager(logConfig).createTailer("debug", name)) {
+            tailer.toStart();
+            displayMessage(tailer, limit, render);
+        }
+    }
+
+    public void cat(String logConfig, String name, int partition, long offset, int limit) throws Exception {
+        cat(logConfig, name, partition, offset, limit, this::renderMessage);
+    }
+
+    public void cat(String logConfig, String name, int partition, long offset, int limit, Consumer<LogRecord<Record>> render) throws Exception {
+        LogPartition part = LogPartition.of(name, partition);
+        LogOffset off = new LogOffsetImpl(part, offset);
+        try (LogTailer<Record> tailer = getLogManager(logConfig).createTailer("debug", part)) {
+            tailer.seek(off);
+            displayMessage(tailer, limit, render);
+        }
+    }
+
+    /**
+     * Tail from the end of a Log
+     */
+    public void tail(String logConfig, String name) throws Exception {
+        tail(logConfig, name, this::renderMessage);
+    }
+
+    public void tail(String logConfig, String name, Consumer<LogRecord<Record>> render) throws Exception {
+        tail(logConfig, name, render, 60);
+    }
+
+    public void tail(String logConfig, String name, Consumer<LogRecord<Record>> render, int timeoutSeconds) throws Exception {
+        String group = "debug";
+        System.out.println("tail on " + name);
```

Comment:
Please don't use `System.out.println ` in any class of `src/main/java`, even though these are debug methods the code checkers will not like it.